sensors/field_odometry.py

```diff
@@ -138,6 +140,29 @@ def keep_pose_in_field(self):
         new_pose = Pose2d(new_x, new_y, pose.rotation())
         self.resetOdometry(new_pose)
 
+    def potential_crash(self):
+        if not config.odometry_crash_detection_enabled:
+            return False
+        accel_x = self.drivetrain.gyro.get_y_accel()
+        accel_y = self.drivetrain.gyro.get_x_accel()
+        
+        rvx = self.drivetrain.chassis_speeds.vx
+        rvy = self.drivetrain.chassis_speeds.vy
+        if (
+            abs(accel_x) > config.odometry_crash_accel_threshold
```

Comment:
I am very concerned about the order of these and/or operations, and to make sure we are getting the result that we really want. Here... I am not sure of the logic behind this... Please explain.